src/checkmatelib/url/canonicalize.py

```diff
@@ -24,12 +24,26 @@ def canonicalize(cls, url):
         :param url: URL to normalise
         :return: A normalised version of the URL
         """
-        parts = cls.canonical_split(url)
-        clean_url = urlunparse(parts)
+        return cls.canonical_join(cls.canonical_split(url))
+
+    @classmethod
+    def canonical_join(cls, parts):
+        """Join canonical parts into a URL.
+
+        This assumes you have called `canonical_split` to get the parts, and
+        does not perform any cleaning itself. The fragment is always ignored,
+        but is present to support the same interface as `urllib`
+
+        :param parts: Tuple of (scheme, netloc, path, params, query, fragment)
```

Comment:
Would it make sense to use a `urllib.parse.ParseResult` for this?  That provides named accessors and implies consistency with `urlparse` results.

Reply:
Yep, worth a shot

Reply:
Done. Works out quite well and makes documentation easier